code/game/machinery/kitchen/snackbarmachine.dm

```diff
@@ -37,39 +54,85 @@
 	user.set_machine(src)
 
 	var/dat = list()
-	if(!beaker)
-		dat += "Please insert a beaker.<BR>"
-	else
+	// Beaker
+	if(beaker)
 		var/datum/reagents/R = beaker.reagents
-		dat += "<A href='?src=\ref[src];eject=1'>Eject beaker and Clear Buffer</A><BR>"
-		if(!R.total_volume)
-			dat += "Beaker is empty."
-		else
-			dat += "Add to buffer:<BR>"
+		dat += "<A href='?src=\ref[src];eject=1'>Eject beaker.</A><BR>"
+
+		if(R.total_volume)
+			// Beaker buttons
+			dat += {"
+				<table>
+					<td class="column1">
+						Add to Snack Buffer: <A href='?src=\ref[src];beaker_addall=1;amount=[R.total_volume]'>All</A>
+					</td>
+				</table>
+			"}
+
+			// Beaker reagents
+			dat += "<table>"
 			for(var/datum/reagent/G in R.reagent_list)
-				dat += {"[G.name] , [G.volume] Units -
-					<A href='?src=\ref[src];analyze=1;desc=[G.description];name=[G.name]'>(Analyze)</A>
-					<A href='?src=\ref[src];add=[G.id];amount=1'>(1)</A>
-					<A href='?src=\ref[src];add=[G.id];amount=5'>(5)</A>
-					<A href='?src=\ref[src];add=[G.id];amount=10'>(10)</A>
-					<A href='?src=\ref[src];add=[G.id];amount=[G.volume]'>(All)</A>
-					<A href='?src=\ref[src];addcustom=[G.id]'>(Custom)</A><BR>"}
-
-		dat += "<HR>Transfer to <A href='?src=\ref[src];toggle=1'>[(!mode ? "disposal" : "beaker")]:</A><BR>"
-		if(reagents.total_volume)
-			for(var/datum/reagent/N in reagents.reagent_list)
-				dat += {"[N.name] , [N.volume] Units -
-					<A href='?src=\ref[src];analyze=1;desc=[N.description];name=[N.name]'>(Analyze)</A>
-					<A href='?src=\ref[src];remove=[N.id];amount=1'>(1)</A>
-					<A href='?src=\ref[src];remove=[N.id];amount=5'>(5)</A>
-					<A href='?src=\ref[src];remove=[N.id];amount=10'>(10)</A>
-					<A href='?src=\ref[src];remove=[N.id];amount=[N.volume]'>(All)</A>
-					<A href='?src=\ref[src];removecustom=[N.id]'>(Custom)</A><BR>"}
-			dat += "<A href='?src=\ref[src];createbar=1'>Create snack bar (10 units max)</A>"
+				dat += "<tr>"
+				dat += {"
+					<td class="column1">
+						[G.name] , [round(G.volume, 0.01)] Units - <A href='?src=\ref[src];analyze=\ref[G]'>(?)</A>
+					</td>
+					<td class="column2">
+						<A href='?src=\ref[src];beaker_add=[G.id];amount=1'>1u</A>
+						<A href='?src=\ref[src];beaker_add=[G.id];amount=5'>5u</A>
+						<A href='?src=\ref[src];beaker_add=[G.id];amount=10'>10u</A>
+						<A href='?src=\ref[src];beaker_addcustom=[G.id]'>Custom</A>
+						<A href='?src=\ref[src];beaker_add=[G.id];amount=[G.volume]'>All</A>
+					</td>
+				"}
+				dat += "</tr>"
+			dat += "</table>"
 		else
-			dat += "Buffer is empty.<BR>"
+			dat += "Beaker is empty."
+	else
+		dat += "No beaker inserted."
+
+	// Buffer - Like normal chem masters, except retains without a beaker.
+	// Makes pills. Can flush or move to internal storage
+	dat += "<HR>"
+	dat += "<b>&ltInternal Snack Buffer&gt</b> <BR>"
+
+	var/b_mode_text
+	if(buffer_mode == 1)
```

Comment:
Use a define for this.
`#define STORAGE 1` and then `if (buffer_mode == STORAGE)`.
Same for next line.

Additionally, you could use a switch statement.